### PR TITLE
Usability: #2145. Added user input request for beef update within 'beef' install script

### PR DESCRIPTION
**Pull request by DeezyE**
# Pull Request

Thanks for submitting a PR! Please fill in this template where appropriate:

## Category
*e.g. Bug, Module, Extension, Core Functionality, Documentation, Tests*

## Feature/Issue Description
**Q:** Please give a brief summary of your feature/fix
**A:**

**Q:** Give a technical rundown of what you have changed (if applicable)
**A:**

## Test Cases
**Q:** Describe your test cases, what you have covered and if there are any use cases that still need addressing.
**A:**

## Wiki Page
*If you are adding a new feature that is not easily understood without context, please draft a section to be added to the Wiki below.*

### CODE REVIEW

**Comment by DeezyE** [comment]
@wheatley this is the PR for the release not my issue

**Comment by bcoles** [comment]
This change is going to hang BeEF at startup every time BeEF is started.

Also, the Ruby interpreter has already loaded `./beef` and other dependencies by the time this executes. If the user selects `y` and updates are applied, will they actually take effect, or will the user need to restart BeEF?

File: beef [hunk 1]
```diff
@@ -31,6 +31,13 @@ if RUBY_PLATFORM.downcase.include?('mswin') || RUBY_PLATFORM.downcase.include?('
   exit 1
 end
 
+#
+# @note Ask user if they would like to update beef
+#
+print "Would you like to check and download the latest BeEF update? y/n: " 
+response = gets
+system("git pull && bundle") if response.strip == 'y'
```
Comment (wheatley) [inline beef:39]: We might not want to use system.
There are two options use the ruby-git gem or use [open3](https://docs.ruby-lang.org/en/2.0.0/Open3.html)

**Review by wheatley — CHANGES_REQUESTED** [review]

**Comment by DeezyE** [comment]
> This change is going to hang BeEF at startup every time BeEF is started.
> 
> Also, the Ruby interpreter has already loaded `./beef` and other dependencies by the time this executes. If the user selects `y` and updates are applied, will they actually take effect, or will the user need to restart BeEF?

Do you mean hang beef at startup because it will wait for user prompt, or is there a greater flaw I've missed?

**Comment by DeezyE** [comment]
I have changed the git command to no longer use 'system'

**Comment by DeezyE** [comment]
@bcoles and @wheatley I have pushed an update which has flags for no prompt update, no prompt skip update, and user prompt to update (the default options) with a 5 second timeout if no input is detected.

File: beef (hunk 1 above)
Comment (DeezyE) [reply beef]: done, removed 'system' good call

**Review by wheatley — APPROVED** [review]
Changes made look good.
Great solution

**Pull request merged**